driver-core/src/main/java/com/datastax/driver/core/policies/DefaultRetryPolicy.java

```diff
@@ -137,6 +138,18 @@ public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int
             : RetryDecision.rethrow();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level, the number of retries or
+     * the possibility that the mutation has been applied server-side.
+     */
+    @Override
+    public RetryDecision onRequestError(Statement statement, ConsistencyLevel cl, DriverException e, int nbRetry) {
```

Comment:
Just saw [the conversation](https://github.com/datastax/java-driver/pull/538#discussion-diff-45864310) on the 2.1 (#538) that we did not change the existing behavior, but for 3.0 it seems more appropriate to rethrow here as you indicated.